src/main/kotlin/app/revanced/patches/all/screenshot/removerestriction/patch/RemoveScreenshotRestrictionPatch.kt

```diff
@@ -1,106 +1,72 @@
 import app.revanced.patcher.annotation.Description
 import app.revanced.patcher.annotation.Name
 import app.revanced.patcher.annotation.Version
-import app.revanced.patcher.data.BytecodeContext
-import app.revanced.patcher.extensions.replaceInstruction
-import app.revanced.patcher.patch.BytecodePatch
-import app.revanced.patcher.patch.PatchResult
-import app.revanced.patcher.patch.PatchResultSuccess
 import app.revanced.patcher.patch.annotations.Patch
 import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.util.patch.AbstractTransformInstructionsPatch
+import app.revanced.util.patch.IMethodCall
+import app.revanced.util.patch.fromMethodReference
 import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.ClassDef
+import org.jf.dexlib2.iface.Method
+import org.jf.dexlib2.iface.instruction.Instruction
 import org.jf.dexlib2.iface.instruction.formats.Instruction35c
 import org.jf.dexlib2.iface.reference.MethodReference
+import java.util.*
+
+private typealias InstructionInfo = Triple<RemoveScreenshotRestrictionPatch.MethodCall, Instruction35c, Int>
 
 @Patch(false)
 @Name("remove-screenshot-restriction")
 @Description("Removes the restriction of making screenshots.")
 @Version("0.0.1")
-class RemoveScreenshotRestrictionPatch : BytecodePatch() {
+class RemoveScreenshotRestrictionPatch : AbstractTransformInstructionsPatch<InstructionInfo>() {
 
     private companion object {
         const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/all/screenshot/removerestriction/RemoveScreenshotRestrictionPatch;"
     }
 
     // Information about method calls we want to replace
-    private enum class MethodCall(
-        val definedClassName: String,
-        val methodName: String,
-        val replacementMethodDefinition: String
-    ) {
+    enum class MethodCall(
+        override val definedClassName: String,
+        override val methodName: String,
+        override val methodParams: Array<String>,
+        override val returnType: String
+    ): IMethodCall {
         SetFlags(
             "Landroid/view/Window;",
             "setFlags",
-            "setFlags(Landroid/view/Window;II)V",
+            arrayOf("I", "I"),
+            "V",
         );
+    }
 
-        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
-            when (this) {
-                SetFlags -> {
-                    method.replaceInstruction(
-                        instructionIndex,
-                        "invoke-static { v${instruction.registerC}, v${instruction.registerD}, v${instruction.registerE} }, ${INTEGRATIONS_CLASS_DESCRIPTOR}->${replacementMethodDefinition}"
-                    )
-                }
-            }
+    override fun filterMap(
+        classDef: ClassDef,
+        method: Method,
+        instruction: Instruction,
+        instructionIndex: Int
+    ): InstructionInfo? {
+        if (classDef.type == INTEGRATIONS_CLASS_DESCRIPTOR) {
```

Comment:
This line is similar to another above, with the exception that you do not remove the semicolon. If this is unexpected, you should create another constant INTEGRATIONS_DESCRIPTOR_START and append the semicolon for the currently used constant in order to not call the method to remove the suffix from the constant every time this method is called. This way you can use either constant for its own purpose.